experimental/packages/opentelemetry-sdk-node/src/TracerProviderWithEnvExporter.ts

```diff
@@ -0,0 +1,118 @@
+import { OTLPTraceExporter as OTLPGrpcTraceExporter} from '@opentelemetry/exporter-trace-otlp-grpc';
+import { ZipkinExporter } from '@opentelemetry/exporter-zipkin';
+import { JaegerExporter } from '@opentelemetry/exporter-jaeger';
+
+export class TracerProviderWithEnvExporters extends NodeTracerProvider {
+  private _configuredExporters: SpanExporter[] = [];
+  private _spanProcessors: SpanProcessor[] | undefined;
+
+  static configureOtlp(): SpanExporter {
+    const protocol = this.getOtlpProtocol();
+
+    switch (protocol) {
+      case 'grpc':
+        return new OTLPGrpcTraceExporter;
+      case 'http/json':
+        return new OTLPHttpTraceExporter;
+      case 'http/protobuf':
+        return new OTLPProtoTraceExporter;
+      default:
+        diag.warn(`Unsupported OTLP traces protocol: ${protocol}. Using http/protobuf.`);
+        return new OTLPProtoTraceExporter;
+    }
+  }
+
+  static getOtlpProtocol(): string {
+    const parsedEnvValues = getEnvWithoutDefaults();
+
+    return parsedEnvValues.OTEL_EXPORTER_OTLP_TRACES_PROTOCOL ??
+          parsedEnvValues.OTEL_EXPORTER_OTLP_PROTOCOL ??
+          getEnv().OTEL_EXPORTER_OTLP_TRACES_PROTOCOL ??
+          getEnv().OTEL_EXPORTER_OTLP_PROTOCOL;
+  }
+
+  protected static override _registeredExporters = new Map<
+    string,
+    () => SpanExporter
+      >([
+        ['otlp', () => this.configureOtlp()],
+        ['zipkin', () => new ZipkinExporter],
+        ['jaeger', () => new JaegerExporter],
+        ['console', () => new ConsoleSpanExporter]
+      ]);
+
+  public constructor(config: NodeTracerConfig = {}) {
+    super(config);
+    let traceExportersList = this.filterBlanksAndNulls(Array.from(new Set(getEnv().OTEL_TRACES_EXPORTER.split(','))));
+
+    if (traceExportersList.length === 0 || traceExportersList[0] === 'none') {
+      diag.warn('OTEL_TRACES_EXPORTER contains "none" or is empty. SDK will not be initialized.');
+    } else {
+      if (traceExportersList.length > 1 && traceExportersList.includes('none')) {
+        diag.warn('OTEL_TRACES_EXPORTER contains "none" along with other exporters. Using default otlp exporter.');
+        traceExportersList = ['otlp'];
+      }
+
+      traceExportersList.forEach(exporterName => {
+        const exporter = this._getSpanExporter(exporterName);
+        if (exporter) {
+          this._configuredExporters.push(exporter);
+        } else {
+          diag.warn(`Unrecognized OTEL_TRACES_EXPORTER value: ${exporterName}.`);
+        }
+      });
+
+      if (this._configuredExporters.length > 0) {
+        this._spanProcessors = this.configureSpanProcessors(this._configuredExporters);
+        this._spanProcessors.forEach(processor => {
+          this.addSpanProcessor(processor);
+        });
+      } else {
+        diag.warn('Unable to set up trace exporter(s) due to invalid exporter and/or protocol values.');
+      }
+    }
+  }
+
+  override register(config?: SDKRegistrationConfig) {
+    if (this._spanProcessors) {
```

Comment:
The original intent of suggesting to override the `addSpanProcessor()` method too and keeping track of calls, was to future-proof this in case this class is ever exported from this package. :slightly_smiling_face:  As `addSpanProcessor()` is `public`, it could be called outside of the constructor. Then registration would not take place when `register()` is called. :thinking: 

This is non-blocking for me though :slightly_smiling_face:

Reply:
That's fine with me. It's a simple change and future-proof like you said. :)